### PR TITLE
Updating TMJP*Check to adopt new interfaces for defining mock values for nodes with checks

### PR DESCRIPTION
**Pull request by yarg0007**
Includes:
* new interfaces
* tests for type identification (needed for export implementation)
* unit tests
* addresses menu bar/item coloring that was not working for Mac OS dark mode

### CODE REVIEW

File: TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockTypeDecoderTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    }
+
+    class StringDeveloperMockType implements DeveloperMockTypeDecoder<String> {
+
+    }
+
+    class UnknownDeveloperMockType implements DeveloperMockTypeDecoder<Object> {
+
+    }
```
Comment (juliandesmarais) [inline TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockTypeDecoderTest.java:73]: private class?

File: TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockListOfValuesTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Test
+    public void getUnknownMockTypeFromGeneric() {
+        DeveloperMockListOfObject unknownDeveloperMockType = new DeveloperMockListOfObject();
+        DeveloperMockType mockType = unknownDeveloperMockType.getMockType();
+        assertThat(mockType, is(equalTo(DeveloperMockType.UNKNOWN)));
+    }
+
+    // Test classes that implement DeveloperMockTypeDecoder with the various primitive box types.
+    class DeveloperMockListOfBoolean implements DeveloperMockListOfValues<Boolean> {
```
Comment (juliandesmarais) [inline TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockListOfValuesTest.java:49]: Should these be marked as private?

**Review by juliandesmarais — APPROVED** [review]
Looks good, one suggestion

File: TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockTypeDecoderTest.java (hunk 1 above)
Comment (yarg0007) [reply TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockTypeDecoderTest.java]: Yup. Good catch.

File: TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockListOfValuesTest.java (hunk 2 above)
Comment (yarg0007) [reply TMBuilder/src/test/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockListOfValuesTest.java]: Yup. Good catch.

**Pull request merged**